core/src/main/java/org/elasticsearch/search/suggest/term/TermSuggester.java

```diff
@@ -56,7 +59,10 @@ public TermSuggestion innerExecute(String name, TermSuggestionContext suggestion
             );
             Text key = new BytesText(new BytesArray(token.term.bytes()));
             TermSuggestion.Entry resultEntry = new TermSuggestion.Entry(key, token.startOffset, token.endOffset - token.startOffset);
-            for (SuggestWord suggestWord : suggestedWords) {
+            if(suggestion.getDirectSpellCheckerSettings().exactMatch()){
+                addExactMatch( suggestion,  indexReader,  token, resultEntry);
```

Comment:
I think we can remove `addExactMatch` method altogether and just use the `TermsEnum#seekExact` to see if the term is present in the term dictionary:
```java
final TermsEnum termsEnum = MultiFields.getTerms(indexReader, token.term.field()).iterator();
if (termsEnum.seekExact(token.term.bytes())) {
    Text word = new StringText(token.term.text());
    resultEntry.addOption(new TermSuggestion.Entry.Option(word, termsEnum.docFreq(), 1f));
}
```